File: solutions/julia/rna-transcription/4/rna-transcription.jl
function to_rna(dna)
      nucleotid_d = Dict('A'=>'U', 'T'=>'A', 'C'=>'G', 'G'=>'C') 
     if isempty(dna)
        return dna
     end
     try 
         map(x -> nucleotid_d[x], dna)
     catch e
        if isa(e,KeyError) 
            throw(ErrorException("invalid input."))
        end
     end   
end

File: solutions/free-pascal/rna-transcription/1/RnaTranscription.pas
unit RnaTranscription;

{$mode ObjFPC}{$H+}

interface

function ToRna(const dna : string) : string;

implementation

uses SysUtils;

function ToRna(const dna : string) : string;
var
  c : string;
  i : integer;
begin
  result := '';
  for i:=1 to length(dna) do
  begin
    case dna[i] of
      'A' : c:='U';
      'T' : c:='A';
      'C' : c:='G';
      'G' : c:='C';
    else
      raise Exception.Create('Invalid Input');
    end;
    result := result + c;
  end;
  

end;

end.
